### PR TITLE
Feat/argent plugin accounts

### PR DESCRIPTION
**Pull request by dhruvkelawala**

### CODE REVIEW

File: packages/extension/src/ui/features/settings/NetworkSettingsFormScreen.tsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ export const NetworkSettingsFormScreen: FC<NetworkSettingsFormScreenProps> = (
               autoComplete="off"
               control={control}
               placeholder="Account class hash"
-              name="accountClassHash"
+              name="accountClassHash.argentAccount"
```
Comment (janek26) [inline packages/extension/src/ui/features/settings/NetworkSettingsFormScreen.tsx:171]: this works? nice

we should add a field to add a plugin account of your choice at some point too

**Review by janek26 — COMMENTED** [review]
nice!

Next up registering the plugin system? :)

File: packages/extension/src/ui/features/settings/NetworkSettingsFormScreen.tsx (hunk 1 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/features/settings/NetworkSettingsFormScreen.tsx:171]: Yeah it works.

Yes, we should. For now, keeping it simple.

**Comment by janek26** [comment]
this needs a starknet.js release before it's ready for review

File: packages/dapp/src/components/TokenDapp.tsx [hunk 2]
```diff
@@ -115,6 +128,57 @@ export const TokenDapp: FC = () => {
...
+      if (!sessionAccount) {
+        throw new Error("No open session")
+      }
+      const erc20Contract = new Contract(
+        Erc20Abi as Abi,
+        getErc20TokenAddress(network),
+        sessionAccount,
+      )
+
+      const result = erc20Contract.mint(
```
Comment (simonheys) [inline packages/dapp/src/components/TokenDapp.tsx:169]:
```suggestion
      const result = await erc20Contract.mint(
```

File: packages/extension/src/shared/account/details/fetchType.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+              [
+                call.contractAddress,
+                hash.getSelectorFromName(call.entrypoint),
+                0,
+              ] as const,
+          )
+
+          const response = await multicallContract.aggregate(multicallCall)
+
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (simonheys) [inline packages/extension/src/shared/account/details/fetchType.ts:75]: Are there unused vars here? Maybe used to be

File: packages/extension/src/shared/settings/index.ts [hunk 4]
```diff
@@ -1,4 +1,7 @@
 export const isPrivacySettingsEnabled =
   (process.env.FEATURE_PRIVACY_SETTINGS || "false") === "true"
 
+export const isExperimentalSettingsEnabled =
+  (process.env.FEATURE_EXPERIMENTAL_SETTINGS || "false") === "true"
```
Comment (simonheys) [inline packages/extension/src/shared/settings/index.ts:5]: I think `FEATURE_EXPERIMENTAL_SETTINGS` needs to be added into `.env.example` too?

File: packages/extension/src/ui/services/useHover.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        node.addEventListener("mouseover", handleMouseOver)
+        node.addEventListener("mouseout", handleMouseOut)
+        return () => {
+          node.removeEventListener("mouseover", handleMouseOver)
+          node.removeEventListener("mouseout", handleMouseOut)
+        }
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [ref.current], // Recall only if ref changes
```
Comment (simonheys) [inline packages/extension/src/ui/services/useHover.ts]: Not sure the intention here, but AFAIK ref or ref.current won't work as dependencies and won't trigger the effect again. There is a solution here https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780 - please let me know if this is wrong as it's a pattern I have used elsewhere

File: packages/extension/src/shared/settings/index.ts (hunk 4 above)
Comment (janek26) [reply packages/extension/src/shared/settings/index.ts:5]: good catch

File: packages/extension/src/ui/services/useHover.ts (hunk 5 above)
Comment (janek26) [reply packages/extension/src/ui/services/useHover.ts]: file removed

File: packages/sessions/README.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
+# Sessions
+
+Sessions can be used to send transactions from a dapp on behalf of a user without requiring his confirmation with a wallet.
```
Comment (simonheys) [inline packages/sessions/README.md:3]:
```suggestion
Sessions can be used to send transactions from a dapp on behalf of a user without requiring their confirmation with a wallet.
```

**Review by simonheys — APPROVED** [review]

**Pull request merged**